### PR TITLE
Header file for Linux-specific timerfd API

### PR DESCRIPTION
**Pull request by adamdruppe**
I want to use this in a project... it appeared about the same time as epoll and has a similarly small header file so I just followed that and ported the C headers.

### CODE REVIEW

**Comment by adamdruppe** [comment]
Why is the test failing?

```
make[2]: execvp: ./obj/linux/32/both: Permission denied
make[2]: *** [obj/linux/32/both.done] Error 127

diff: ./obj/freebsd/32/bothgc.log: No such file or directory
[snip]
```

I try not to curse anymore, so I think I had best just walk away for tonight.

File: src/core/sys/linux/timerfd.d [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * D header file to interface with the Linux timefd API <http://man7.org/linux/man-pages/man2/timerfd_create.2.html>
+ * Available since Linux 2.6
+ *
+ * License : $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ */
+module core.sys.linux.timerfd;
+
+version (linux):
+
+public import core.sys.posix.time;
```
Comment (DmitryOlshansky) [inline src/core/sys/linux/timerfd.d]: public imports are dangerous ;) There is at least one bug with them still open

Comment (adamdruppe) [inline src/core/sys/linux/timerfd.d]: On Sat, Sep 12, 2015 at 03:41:57AM -0700, Dmitry Olshansky wrote:
> public imports are dangerous ;) There is at least one bug with them still open

Perhaps we should fix that bug!

I could make it private, but the file is basically useless without that
itimerspec definition, and the file is straight up #included in the C
source so bugs notwithstanding, the public import is the closest match.

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**